ext/IPC-Open3/lib/IPC/Open3.pm

```diff
@@ -208,8 +208,40 @@ sub _open3 {
 	croak "$Me: $@";
     }
 
-    my @handles = ({ mode => '<', handle => \*STDIN },
-		   { mode => '>', handle => \*STDOUT },
+    my $stdin;
+    my $stdout;
+
+    # The logic below is complicated.
+    # We want a Perl filehandle that refers to POSIX file descriptor 0.
+    # That's a prerequisite for setting up pipes correctly.  So first,
+    # we try to fdopen(0).  If file descriptor 0 is closed, however,
+    # the fdopen will fail.  So then we open a filehandle for input
+    # from /dev/null, and verify that we do indeed get fd 0 back.  It
+    # doesn't particularly matter that this filehandle is open for
+    # input from /dev/null because it will be converted to refer to
+    # the child end of the pipe anyway.  However, /dev/null is UNIX-specific
+    # and I have no access to non-UNIX systems to make it portable.
+    unless (open($stdin, '<&=', 0)) {
+            unless (open($stdin, '<', '/dev/null')) {
```

Comment:
That is a unix assumption. `File::Spec->devnull` is the portable alternative.

Reply:
Thanks; fixed.